_posts/en/newsletters/2023-04-19-newsletter.md

```diff
@@ -0,0 +1,210 @@
+  involves the creation and transfer of tokens, although RGB is designed
+  for more purposes than just token transfer.
+
+  - Offchain, Alice creates a contract whose initial state assigns 1,000
+    tokens to a certain UTXO which she controls.
+
+  - Bob wants 400 of the tokens.  Alice gives him a copy of the original
+    contract plus a transaction which spends her UTXO a new output.
+    That output contains a non-public commitment to the new contract
+    state.  The new contract state specifies the distribution of amounts
+    (400 to Bob; 600 back to Alice) and the identifiers for two outputs
+    which will control those amounts.  The security of this token
+    transfer is now equal to that of Alice's Bitcoin transaction, e.g.
+    when her transaction has six confirmations, the token transfer will
+    be secure against a fork of up to six blocks.
+
+      The outputs which control the amounts don't need to be outputs of
+      the transaction containing the commitment (though that would be
+      allowed).  The bitcoin value of the various outputs, and their
+      other characteristics, are irrelevant to the RGB protocol,
+      although Alice and Bob will want to ensure they're easy to spend.
+
+  - Later, Carol wants to buy 100 tokens from Bob in an onchain atomic
+    swap.  She generates an unsigned PSBT which funds the transaction
+    from her inputs, pays Bob via an output, and returns the bitcoin
+    change to herself with a second output.  She also tells Bob the
+    identifier for the UTXO where she wants to receive her tokens.
+
+      Bob provides Carol with the original contract and the commitment
+      Alice previously created that proves Bob now controls 400 tokens.
+      He updates the PSBT with his UTXO controlling the tokens as an
+      input and adds an output which commits to the new contract
+      state---but just the part of it involving him and Carol.  He
+      doesn't need to know what Alice did with her remaining 600 tokens
+      and Alice has no involvement in the exchange between Bob and
+      Carol.  Bob completes his portion of the PSBT by providing his
+      signature.
+
+      Alice verifies the original contract and the history of previous
+      state updates.  She also ensures that the output in the PSBT
+      commits to giving her 100 tokens (and is otherwise correct).  She
+      provides her signature and broadcasts the transaction.
+
+  Although each of the example transfers above was made onchain, it's
+  straightforward to modify the protocol to operate offchain.  Carol
+  gives a copy of the contract to Dan along with the history of state
+  updates leading to her receiving 100 tokens.  She and Dan then
+  coordinate to create an output which receives the 100 tokens and that
+  requires signatures from both of them to spend.  Offchain, they
+  transfer the tokens back and forth by generating many different
+  versions of a transaction that spends the multisignature output, with
```

Comment:
```suggestion
  versions of a transaction that spends the [multisignature][topic multisignature] output, with
```

Reply:
Skipping this suggestion.  Our topic about "multisignature" only refers to non-script-based multisignatures, whereas in the above text it doesn't matter whether we use script-based multisignatures (as used in LN now) or non-script-based multisignatures (future).

While writing about the subject for Mastering Bitcoin, I came up with some new terms which passed the sniff test on bitcoin-wizards, so I'll try to get around to updating our topics and style pages for that.